### ISSUE TITLE
Allow custom username and password for each resource type

### ISSUE THREAD

**Issue by cliffano** [shinesolutions/puppet-aem-resources#5]
AEM username and password are currently only configurable via environment variable and config file.
Need to add mechanism to override them at resource type parameter level.

**Comment by henrykuijpers** [comment]
Indeed! This would be very helpful!

Trying to use the module, but I'm setting up 2 instances on one machine (for development purposes). I don't think I would be able to have different configuration for the 2 machines, would I?

**Comment by cliffano** [comment]
@henrykuijpers At the moment, the only way to set up multiple AEM instances on one machine is if you either:
- pass configuration via environment variables when you run puppet:
```
aem_username=myadmin1 aem_password=mypassword1 aem_host=localhost aem_port=4502 aem_protocol=http puppet apply manifest1.pp
aem_username=myadmin2 aem_password=mypassword2 aem_host=localhost aem_port=4503 aem_protocol=https puppet apply manifest2.pp
```
- or by setting two AEM config YAML files, one for each AEM instance, and then set the one you're currently provisioning as `aem.yaml` in Puppet config path

I think what would be very useful for your scenario is the ability to have multiple AEM config files and then set it once per `puppet apply` call. i.e. don't restrict it to `aem.yaml` .

**Comment by cliffano** [comment]
@henrykuijpers Multi aem instances support has been added to master (to be released as part of 1.3.0). It's currently being tested by a number of users. Ticket #19 https://github.com/shinesolutions/puppet-aem-resources/issues/19
Please check out the doco https://github.com/shinesolutions/puppet-aem-resources#multi-aem-instances .

**Comment by cliffano** [comment]
Done. Will be included in 2.0.0 ."
